WordPress/gutenberg
# Integrate the mobile React Native testsuite with the main Gutenberg build

## Description
This PR adds the [native mobile app repo](https://github.com/wordpress-mobile/gutenberg-mobile)  as a git submodule and includes the mobile testsuite into the `npm test` script and the Travis test pipeline.

## How has this been tested?
Locally doing `npm test` runs green.

### To test:
1. Checkout the Gutenberg code into a new directory and `cd` into it
2. `git submodule update --init --recursive` to pull in the mobile submodule and its submodules
3. Follow the typical Gutenberg setup instructions
4. Tests should run green and also include the mobile tests. To confirm, look for the `Setting RN platform to` string in your terminal's ouput.

## Types of changes
This PR doesn't introduce any code changes. All changes are done in the setup/tooling side of the project. In particular, the changes include:

* Installing `yarn`, the package manager used by the mobile project. That's needed by the mobile npm scripts only.
* The mobile repo has its own set of linting, compilation and run configurations so all those are ignored. `.eslintignore` got adjusted and `.stylelinignore` is introduced.
* The `test` npm script adopts the `test-mobile` script to run after the Gutenberg unit tests
* 3 jobs are added to Travis: code correctness tests, Android codepath tests, iOS codepath tests
* Gutenberg testsuites (unit, e2e) are ignoring the test modules inside the mobile folder since those need their own Jest configuration.

## Checklist:
